src/vivarium_nih_us_cvd/components/treatment.py

```diff
@@ -240,6 +246,121 @@ def _register_target_modifiers(self, builder: Builder) -> None:
             ],
         )
 
+        # drop value modifiers
+        builder.value.register_value_modifier(
+            data_values.PIPELINES.BMI_DROP_VALUE,
+            modifier=self._apply_lifestyle_to_bmi,
```

Comment:
Consider doing something with partial functions like this so you don't have to define 3 functions that are almost identical:
```
def _apply_lifestyle(self, index: pd.Index, target: pd.Series, risk: str):
    # allow for updating drop value of dead people - makes interacting with target easier
    pop = self.population_view.get(index)
    enrollment_dates = pop[data_values.COLUMNS.LIFESTYLE]
    updated_drop_values = self.get_updated_drop_values(
        target, enrollment_dates, risk=risk
    )
    
return updated_drop_values
```

And when registering the value modifier you can do this:
```
builder.value.register_value_modifier(
    dava_values.PIPELINES.BMI_DROP_VALUE,
    modifier=partial(self._apply_lifestyle, "bmi"),
   ...
)
```

Reply:
Unfortunately this returns the following error

```
File ~/repos/vivarium_nih_us_cvd/src/vivarium_nih_us_cvd/components/treatment.py:250, in Treatment._register_target_modifiers(self, builder)
    240 builder.value.register_value_modifier(
    241     data_values.PIPELINES.LDLC_EXPOSURE,
    242     modifier=self.ldlc_target_modifier,
   (...)
    246     ],
    247 )
    249 # drop value modifiers
--> 250 builder.value.register_value_modifier(
    251     data_values.PIPELINES.BMI_DROP_VALUE,
    252     modifier=partial(self._apply_lifestyle, risk="bmi"),
    253     requires_columns=[
    254         data_values.COLUMNS.LIFESTYLE,
    255     ],
    256 )
    258 builder.value.register_value_modifier(
    259     data_values.PIPELINES.FPG_DROP_VALUE,
    260     modifier=partial(self._apply_lifestyle, risk="fpg"),
   (...)
    263     ],
    264 )
    266 builder.value.register_value_modifier(
    267     data_values.PIPELINES.SBP_DROP_VALUE,
    268     modifier=partial(self._apply_lifestyle, risk="sbp"),
   (...)
    271     ],
    272 )

File ~/repos/vivarium/src/vivarium/framework/values.py:626, in ValuesInterface.register_value_modifier(self, value_name, modifier, requires_columns, requires_values, requires_streams)
    592 def register_value_modifier(
    593     self,
    594     value_name: str,
   (...)
    598     requires_streams: List[str] = (),
    599 ):
    600     """Marks a ``Callable`` as the modifier of a named value.
    601 
    602     Parameters
   (...)
    624 
    625     """
--> 626     self._manager.register_value_modifier(
    627         value_name, modifier, requires_columns, requires_values, requires_streams
    628     )

File ~/repos/vivarium/src/vivarium/framework/lifecycle.py:384, in ConstraintMaker.constrain_normal_method.<locals>._wrapped(*args, **kwargs)
    380 self.check_valid_state(method, permitted_states)
    381 # Call the __func__ because we're rebinding _wrapped to the method
    382 # name on the object.  If we called method directly, we'd get
    383 # two copies of self.
--> 384 return method.__func__(*args, **kwargs)

File ~/repos/vivarium/src/vivarium/framework/values.py:386, in ValuesManager.register_value_modifier(self, value_name, modifier, requires_columns, requires_values, requires_streams)
    352 def register_value_modifier(
    353     self,
    354     value_name: str,
   (...)
    358     requires_streams: List[str] = (),
    359 ):
    360     """Marks a ``Callable`` as the modifier of a named value.
    361 
    362     Parameters
   (...)
    384 
    385     """
--> 386     modifier_name = self._get_modifier_name(modifier)
    388     pipeline = self._pipelines[value_name]  # May create a pipeline
    389     pipeline.mutators.append(modifier)

File ~/repos/vivarium/src/vivarium/framework/values.py:447, in ValuesManager._get_modifier_name(modifier)
    445     modifier_name = modifier.__name__
    446 elif hasattr(modifier, "__call__"):  # Some anonymous callable
--> 447     modifier_name = f"{modifier.__class__.name__}.__call__"
    448 else:  # I don't know what this is.
    449     raise ValueError(f"Unknown modifier type: {type(modifier)}")

AttributeError: type object 'functools.partial' has no attribute 'name__'
```

Reply:
In that case, retain the `self._apply_lifestyle` function and define three functions like 
```
def _apply_lifestyle_to_bmi(self, index: pd.Index, target: pd.Series):
    return self._apply_lifestyle(index, target, "bmi")
```